CMakeLists.txt

```diff
@@ -59,6 +58,14 @@ else()
     set(HAVE_SYMENGINE_LLVM False)
 endif()
 
+if(WITH_NUMPY)
+    find_package(NumPy REQUIRED)
+    include_directories(${NUMPY_INCLUDE_PATH})
```

Comment:
If so, I think we can just use `numpy.ndarray.ctypes` to get a array pointer and have numpy only as a runtime requirement. I'll have a look to see if this is possible.